packages/flutter_tools/lib/src/commands/build_web.dart

```diff
@@ -59,6 +60,11 @@ class BuildWebCommand extends BuildSubCommand {
                        'is not desirable',
       },
     );
+    argParser.addOption('base-href',
+    defaultsTo: '/',
```

Comment:
Setting a default value means we'll always override what's in `index.html`.

Could we remove this default, so that when `base-href` is omitted, we keep whatever base href that already exists in `index.html`?

Reply:
removed the default value  . 
Now before calling the addBaseHrefToIndexPage function , we check if the baseHref value is not null . if it is null , no changes  are done to the index.html , if it is not null , then we change the base tag href's value to the given value .